pkg/sidecars/server/handlers.go

```diff
@@ -0,0 +1,136 @@
+package sidecarserv
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/inspr/inspr/pkg/environment"
+	"github.com/inspr/inspr/pkg/ierrors"
+	"github.com/inspr/inspr/pkg/rest"
+	"go.uber.org/zap"
+)
+
+var logger *zap.Logger
+
+func init() {
+	logger, _ = zap.NewProduction()
+}
+
+const maxBrokerRetries = 5
+
+var (
+	writeMessageErr = ierrors.NewError().InternalServer().Message("broker's writeMessage failed")
+)
+
+// handles the /message route in the server
+func (s *Server) writeMessageHandler() rest.Handler {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logger.Info("handling message write")
+
+		channel := strings.TrimPrefix(r.URL.Path, "/")
+
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			rest.ERROR(w, err)
+			return
+		}
+
+		if !environment.OutputBrokerChannnels(s.broker).Contains(channel) { // OutputChannnelList must be checked for obtaining the right list
+			insprError := ierrors.
+				NewError().
+				BadRequest().
+				Message(
+					"channel '%s' not found",
+					channel,
+				)
+
+			rest.ERROR(w, insprError.Build())
+			return
+		}
+
+		logger.Info("writing message to broker", zap.String("channel", channel))
+		if err := s.Writer.WriteMessage(channel, body); err != nil {
+			rest.ERROR(w, writeMessageErr.InnerError(err).Build())
```

Comment:
As it is now, everything put inside `InnerError` is not being returned/shown, so it makes it really hard to understand what happened. You should probably just return an ierror contaninig the `err` as it's message or something like this.